test/sphere_Plm.jl

```diff
@@ -0,0 +1,143 @@
+            if x(col) < 0
+                P[mm1, col] = sign(rem[n + 1, 2] - 0.5) * tstart
+            end
+
+            # Recur from m1 to m = 0, accumulating normalizing factor.
+            sumsq = tol
+            for m in (mm1 - 2):-1:0
+                P[m + 1, col] =
+                    (
+                        (m + 1) * twocot[col] * P[m + 2, col] -
+                        rootn[n + m + 3] * rootn[n - m] * P[m + 3, col]
+                    ) / (rootn[n + m + 2] * rootn[n - m + 1])
+                sumsq = P[m + 1, col]^2 + sumsq
+            end
+            scale = 1 / sqrt(2 * sumsq - P[1, col]^2)
+            P[1:(mm1 + 1), col] = scale * P[1:(mm1 + 1), col]
+        end     # FOR loop
+    end # small sine IF loop
+
+    # Find the values of x,s for which there is no underflow, and for
+    # which twocot is not infinite (x~=1).
+
+    nind = findall((x .!= 1) .& (abs.(sn) .>= tol))
+    if ~isempty(nind)
+
+        # Produce normalization constant for the m = n function
+        d = 2:2:(2 * n)
+        c = prod(ones(n, 1) - 1.0 ./ d)
+
+        # Use sn = (-s).^n (written above) to write the m = n function
+        P[n + 1, nind] = sqrt(c) * sn[nind]
+        P[n, nind] = P[n + 1, nind] .* twocot[nind] * n ./ rootn[end]
+
+        # Recur downwards to m = 0
+        for m in (n - 2):-1:0
+            P[m + 1, nind] =
+                (
+                    P[m + 2, nind] .* twocot[nind] * (m + 1) -
+                    P[m + 3, nind] * rootn[n + m + 3] * rootn[n - m]
+                ) / (rootn[n + m + 2] * rootn[n - m + 1])
+        end
+    end # IF loop
+
+    y = P[1:(n + 1), :]
+
+    # Polar argument   (x = +-1)
+    s0 = findall(s == 0)
+    if ~isempty(s0)
+        y[1, s0] = x[s0] .^ n
+    end
+
+    for m in 1:(n - 1)
+        y[m + 1, :] = prod(rootn[(n - m + 2):(n + m + 1)]) * y[m + 1, :]
+    end
+    # Last row (m = n) must be done separately to handle 0!.
+    # NOTE: the coefficient for (m = n) overflows for n>150.
+    y[n + 1, :] = prod(rootn[2:end]) * y[n + 1, :]
+
+
+    # Restore original dimensions
+    if length(sizex) > 2 || minimum(sizex) > 1
+        y = reshape(y, (n + 1, sizex...))
+    end
+    return y
+end
+
+function harmonicVec(n, m, lat, lon, r)
+
+    isoddm = mod(m, 2) == 1
+    isnegm = m < 0
+
+    m = abs(m)
+    C = 1 / sqrt(2 * pi)
+
+    P = legendre(n, sin.(lat))
+    V =
+        0.5 *
+        (P[abs(m + 1) + 1, :] - (n + m) * (n - m + 1) * P[abs(m - 1) + 1, :]) /
+        sqrt(n * (n + 1))
+
+    P = legendre(n - 1, sin.(lat))
+
+    W =
+        0.5 *
+        (P[abs(m + 1) + 1, :] + (n + m) * (n + m - 1) * P[abs(m - 1) + 1, :]) /
+        sqrt(n * (n + 1))
+    E = exp.((im * m) .* lon)
+
+    if isnegm
+        E = conj(E)
+        if isoddm
+            E = -E
+        end
+    end
+    fu = real.(C * ((im * W) - V) .* E)
+    fv = real.(C * (V + (im * W)) .* E)
+    # fu = real.(C * ((im * W)) .* E)
+    # fv = real.(C * (V) .* E)
```

Comment:
Sorry, commented code is still here. It's better not to merge stale code/comments. It probably slipped if you rebased

Reply:
@valeriabarra  Thanks for keeping an eye here! It is not fixed yet, the uncommented lines are simplified versions of these two commented lines and both work fine in this test. I need to double check with @OsKnoth and keep one in the end.

Reply:
Thanks for clarifying :)

Reply:
This file was removed. The new file "sphere_sphericalharmonics.jl" contains both functions. The commented codes are kept intentionally because that package is preferred but only has personal registry. The plan is to switch to that package once the author have it in General registry.